packages/google_sign_in/google_sign_in/example/lib/main.dart

```diff
@@ -38,20 +42,36 @@ class SignInDemo extends StatefulWidget {
 
 class SignInDemoState extends State<SignInDemo> {
   GoogleSignInAccount? _currentUser;
+  bool _isAuthorized = false; // has granted permissions?
   String _contactText = '';
 
   @override
   void initState() {
     super.initState();
-    _googleSignIn.onCurrentUserChanged.listen((GoogleSignInAccount? account) {
+
+    _googleSignIn.onCurrentUserChanged
+        .listen((GoogleSignInAccount? account) async {
+      // Check if the account can access scopes...
+      bool isAuthorized = false;
+      if (account != null) {
+        final String? accessToken = (await account.authentication).idToken;
+        isAuthorized = await _googleSignIn.canAccessScopes(accessToken, scopes);
+      }
       setState(() {
         _currentUser = account;
+        _isAuthorized = isAuthorized;
       });
-      if (_currentUser != null) {
-        _handleGetContact(_currentUser!);
+      if (isAuthorized) {
+        _handleGetContact(account!);
       }
     });
-    _googleSignIn.signInSilently();
+
+    // Ensure the plugin is fully setup (calls _ensureInitialized)
+    if (kIsWeb) {
+      _googleSignIn.isSignedIn();
```

Comment:
do we need to wait here due to?
https://github.com/flutter/packages/blob/8285a64e9444b23539e69b2253ab60b9e1e4f5a4/packages/google_sign_in/google_sign_in/lib/google_sign_in.dart#L399

Reply:
We can't await in the `initState`. The only way I found this to 100% prevent a race condition is something like:

_googleSignIn.isSignedIn().then((_) {
  // plugin-dependent initialization, like using the stream.
});

Attaching the stream *before* calling isSignedIn works, BUT calling isSignedIn right before using the stream makes (my test) hang.

Reply:
So this is called to make sure that when the render button is used, an initialized client is available?

Reply:
@wreppun Yes, since `renderButton` is a web-package-only thing, this needs to be called so the core plugin calls `_ensureInitialized`, which is when the web clients get initialized, and we can connect the Event stream from the web package into the core package.

An alternative to this would be to expose a `Future<Widget> renderSignInButton(Options)` through the core plugin that could internally `await _ensureInitialized`.